<launch>

  <arg name="data_directory" default="$(find bwi_kr)/config/real3ne_all_rooms" />
  <arg name="map_file"  default="$(find bwi_kr)/maps/map_whole2_with_elevators_new_forbidden.yaml" />

  <!--Multi map node-->
  <include file="$(find bwi_kr)/launch/multi_map_elevator_demo.launch"> </include>

  <!-- launch real world robot with map file and navigation -->
  <include file="$(find segbot_navigation)/launch/navigation.launch">
    <arg name="map_file" value="$(arg map_file)" />
  </include>

  <!-- launch logical navigator -->
  <node name="segbot_logical_navigator" pkg="segbot_logical_translator"
        type="segbot_logical_navigator" output="screen">
    <param name="map_file" value="$(arg map_file)" />
    <param name="data_directory" value="$(arg data_directory)" />
  </node>
  
  <!-- launch PCL people detection node -->
  <include file="$(find pcl_perception)/launch/background_people_detection.launch">
  </include>

  <!-- launch rosbridge_server for remote video streaming -->
  <include file="$(find rosbridge_server)/launch/rosbridge_websocket.launch"> 
  </include>

  <!-- launch mjpeg_server for remote video streaming -->
  <node name="mjpeg_server" pkg="mjpeg_server" type="mjpeg_server" 
        output="screen">
    <param name="__image_transport" value="compressed" />
  </node>

  <!-- launch the gui -->
  <node name="segbot_gui" pkg="segbot_gui" type="question_dialog_plugin" />

  <!-- launch the sound play node -->
  <node name="soundplay" pkg="sound_play" type="soundplay_node.py" />

  <!-- logging node -->
  <include file="$(find bwi_logging)/launch/record.launch"> </include>



</launch>
